src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -350,6 +350,53 @@ static Value builtinRegExpSplit(ExecutionState& state, Value thisValue, size_t a
     return A;
 }
 
+#ifdef ESCARGOT_ENABLE_ES2015
+static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(rx, Object, match);
+
+    String* str = argv[0].toString(state);
+    ASSERT(str != nullptr);
+
+    //21.2.5.6.8
+    bool global = rx->get(state, ObjectPropertyName(state, state.context()->staticStrings().global)).value(state, rx).toBoolean(state);
+
+    if (global == false)
+        return builtinRegExpExec(state, rx->asRegExpObject(state), 1, &argv[0], true);
+
+    bool fullUnicode = rx->get(state, ObjectPropertyName(state, state.context()->staticStrings().unicode)).value(state, rx).toBoolean(state);
+    rx->setThrowsException(state, ObjectPropertyName(state, state.context()->staticStrings().lastIndex), Value(0), rx);
+    ArrayObject* A = new ArrayObject(state);
+    size_t n = 0;
+
+    //21.2.5.6.8.g.i
+    while (true) {
+        //21.2.5.6.8.g.i
+        Value result = builtinRegExpExec(state, rx->asRegExpObject(state), 1, &argv[0], true);
+
+        //21.2.5.6.8.g.iii
+        if (result.isNull() == true) {
+            if (n == 0) {
+                return Value(Value::Null);
+            } else {
+                return A;
+            }
+        } else {
+            //21.2.5.6.8.g.iv
+            Value matchStr = result.asObject()->get(state, ObjectPropertyName(state, Value(0))).value(state, result);
```

Comment:
`Let matchStr be ToString(Get(result, "0")).` I think the toString call is missing.